Cog 1273 enhance unittests on pycognaize table_field

AniZakaryan2001 opened the pull request.

AniZakaryan2001 committed:
Code:

Handle exception case for construct_from_raw function in AreaField

AniZakaryan2001 committed:
Code:

* add unittest for date_field

AniZakaryan2001 committed:
Merge branch 'unittests' of https://github.com/cognaize/pycognaize into COG-1273-Enhance-unittests-on-Pycognaize

# Conflicts:
#	pycognaize/tests/test_document/test_field/test_area_field.py
#	pycognaize/tests/test_document/test_field/test_date_field.py

AniZakaryan2001 committed:
Code:

*Add XBRL document.json
*Create HTMLTableTag

AniZakaryan2001 committed:
Code:

*Add XBRL document.json
*Create HTMLTableTag

AniZakaryan2001 committed:
Code:

*create HTMLTableTag
*Create TableField object which attribute is the HTMLTableTag
*Add unittest for get_table_title() function

AniZakaryan2001 committed:
Code:

*Add unittest for test_get_table_title

AniZakaryan2001 committed:
Code:

*Add document id

AniZakaryan2001 committed:
Code:

*Removed comments and blank lines

AniZakaryan2001 committed:
Code:

*Deleted blank lines

AniZakaryan2001 committed:
File:

* add files for unittests

hovhannes-zohrabyan commented on pycognaize/tests/test_common/test_field_collection.py:
```diff
@@ -5,8 +5,12 @@
+
 from collections import defaultdict
 
+
```
Remove the blank line please

hovhannes-zohrabyan commented on pycognaize/tests/test_common/test_field_collection.py:
```diff
@@ -5,8 +5,12 @@
 import uuid
+
+from copy import deepcopy
+
```
this blank line also

hovhannes-zohrabyan commented on pycognaize/tests/test_document/test_field/test_area_field.py:
```diff
@@ -49,14 +50,34 @@ def test__construct_from_raw(self):
+            mock_debug.assert_called_once_with("Failed creating tag for field 5e41a093f4b20400137938d9: 'left'")
+
+
+        # with self.assertRaises(KeyError):
```
Please remove the commented code

hovhannes-zohrabyan commented on pycognaize/tests/test_document/test_field/test_area_field.py:
```diff
@@ -106,4 +127,4 @@ def test___str__(self):
 
     def test_confidence(self):
-        self.assertAlmostEqual(self.area_field_with_tags.confidence, -1.0)
+        self.assertAlmostEqual(self.area_field_with_tags.confidence, -1.0)
```
Add blank line at the end of the line

hovhannes-zohrabyan commented on pycognaize/tests/test_document/test_field/test_date_field.py:
```diff
@@ -113,4 +120,4 @@ def test_group_key(self):
         to_dict_keys = [IqFieldKeyEnum.name.value, IqFieldKeyEnum.data_type.value, ID,
                         IqFieldKeyEnum.tags.value, IqFieldKeyEnum.group_key.value, IqFieldKeyEnum.value.value]
-        self.assertEqual(sorted(test_dict.keys()), sorted(to_dict_keys))
+        self.assertEqual(sorted(test_dict.keys()), sorted(to_dict_keys))
```
Ad blank line

hovhannes-zohrabyan commented on pycognaize/tests/test_document/test_field/test_table_field.py:
```diff
@@ -65,8 +71,28 @@ def setUp(self):
+
+    def test_get_table_title(self):
+        title_1 = self.tbl_field_2.get_table_title(n_lines_above=15, margin=9)
+        self.assertEqual(title_1, '\u200b (exact name of registrant as specified in its charter):usa truck inc.')
```
Please strip \\u200b from the output and from the expected string and check assert equals

hovhannes-zohrabyan commented on pycognaize/tests/test_document/test_field/test_table_field.py:
```diff
@@ -54,8 +58,10 @@ def setUpClass(cls) -> None:
 
     def setUp(self):
         self.doc_id = '60b76b3d6f3f980019105dac'
+        self.doc_id2 = '646c8efb96bed200112575e8'
```
Please use python naming convention

hovhannes-zohrabyan commented on pycognaize/tests/test_document/test_field/test_text_field.py:
```diff
@@ -113,4 +124,4 @@ def test_group_key(self):
         to_dict_keys = [IqFieldKeyEnum.name.value, IqFieldKeyEnum.data_type.value, ID,
                         IqFieldKeyEnum.tags.value, IqFieldKeyEnum.group_key.value, IqFieldKeyEnum.value.value]
-        self.assertEqual(sorted(test_dict.keys()), sorted(to_dict_keys))
+        self.assertEqual(sorted(test_dict.keys()), sorted(to_dict_keys))
```
Add blank line at the end of file

AniZakaryan2001 committed:
Code:
*Create ExtractionTag object and define path for that object
*Add unittest for test_get_table_title_from_pdf

AniZakaryan2001 committed:
Code:
*Create ExtractionTag object and define path for that object
*Add unittest for test_get_table_title_from_pdf

AniZakaryan2001 committed:
Code:
*removed blank lines

AniZakaryan2001 committed:
Code:
*Add unittest for test_construct_from_raw() function

AniZakaryan2001 committed:
Code:
* Stripped \u200b from the output and from the expected string and checked assert equals

AniZakaryan2001 committed:
Code:
* removed blank lines and unused imports

AniZakaryan2001 committed:
Code:
* removed blank lines

AniZakaryan2001 committed:
Code:
* removed blank lines
